table/index_builder.cc

```diff
@@ -24,21 +25,22 @@ namespace rocksdb {
 IndexBuilder* IndexBuilder::CreateIndexBuilder(
     BlockBasedTableOptions::IndexType index_type,
     const InternalKeyComparator* comparator,
+    const SliceTransform* slice_transform,
     const SliceTransform* prefix_extractor, int index_block_restart_interval,
-    uint64_t index_per_partition) {
+    uint64_t index_per_partition, const BlockBasedTableOptions& table_opt) {
   switch (index_type) {
     case BlockBasedTableOptions::kBinarySearch: {
       return new ShortenedIndexBuilder(comparator,
                                        index_block_restart_interval);
     }
     case BlockBasedTableOptions::kHashSearch: {
-      return new HashIndexBuilder(comparator, prefix_extractor,
+      return new HashIndexBuilder(comparator, slice_transform,
                                   index_block_restart_interval);
     }
     case BlockBasedTableOptions::kTwoLevelIndexSearch: {
-      return new PartitionIndexBuilder(comparator, prefix_extractor,
-                                       index_per_partition,
-                                       index_block_restart_interval);
+      return PartitionedIndexBuilder::CreateIndexBuilder(
+          comparator, prefix_extractor, index_block_restart_interval,
+          index_per_partition, table_opt);
```

Comment:
yeah i had to add table_opt later in the coding. yeah it makes sense to remove the two parameters.